operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -752,23 +752,25 @@ func (r *StackReconciler) Reconcile(ctx context.Context, request ctrl.Request) (
 		}
 
 		// Requeue reconciliation as necessary to detect branch updates and
-		// resyncs.
-
+		// resyncs. The logic finds the smallest requeue time (if any) among various polling loops.
 		requeueAfter := time.Duration(0)
 
+		// Try again with exponential backoff if the update failed.
 		if instance.Status.LastUpdate.State == shared.FailedStackStateMessage {
 			requeueAfter = max(1*time.Second, time.Until(instance.Status.LastUpdate.LastResyncTime.Add(cooldown(instance))))
 		}
-		if sess.stack.ContinueResyncOnCommitMatch {
+		// Schedule another poll if ContinueResyncOnCommitMatch is set, for drift detection or to maintain dynamic resources.
+		if instance.Status.LastUpdate.State == shared.SucceededStackStateMessage && sess.stack.ContinueResyncOnCommitMatch {
```

Comment:
The "ContinueResyncOnCommitMatch" polling loop should activate only when the last update was successful.  If failed, the backoff loop should prevail (and is likely a shorter interval).